rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+
+<details>
+<summary>Arguments</summary>
+
+- (+) Doc-comments should have only one variant to reduce complexity. Referencing the next node seems straightforward. 
+- (-) A variant that references the previous node in the AST should be avoided to reduce complexity.
+- (+) Relation between documentation and the referenced implementation is straightforward and back-trackable.
+- (-) Concrete Implementation might be complex.
+- (-) Unclear if complete documentation might also need backward references. 
+    - (-), e.g., rust uses backward references only at the top of file comments.
+    - (+) Nix files can only have ONE expression; the next AST Node, in case of top-of-file comments, is thus always only that one expression. (Unlike in Rust)
+- (-) Tools need to be smart enough to understand asignments `=` and other forms of creating names for anonymous expressions. (e.g., `callPackage` and `import` )
+    - (+) Tools can still come up with other solutions that do not involve calculating everything dynamically from nix code but could also involve a static configuration.
+    - (+) The whole `tool point` is an implementation detail as long as it is not impossible. The current tool, `nixdoc`, already proves that it is possible to have static documentation to a certain degree.
+
+</details>
+
+## Single-line doc-comments (do not exist)
+
+**Observing**: Nix offers two variants of comments; single- (`#`) and multi-line comments (`/* */`). There may be use cases where it is desirable to have a form of single-line comments subtyped for doc-comment purposes. 
+
+**Considering**: Single-line comment for documentation.
+
+**Decision**: Single-line comments **cannot be used** in any form for documentation puposes.
+
+<details>
+<summary>Arguments</summary>
+
+- (-) Doc-comments should have only one variant to reduce complexity.  
+- (-) documentation will likely take up more than one line.
+- (-) If documentation grows bigger than one line, refactoring into a multiline-doc-comment must occur.
+- (+) It Would be consistent with providing variants for both nix comments.
+- (+) Offer the choice.
+- (o) Single lines can also be concatenated to form multi-line documentation.
+- (+) Takes up less vertical space
+- (-) Visually confusing when every line starts with a reserved character.
+    - (-) Potential visual conflicts with the content
+- (+) Indentation of the content is clear.
+
+</details>
+
+## Doc-comment examples
+
+The following examples demonstrate the concrete usage scenarios:
+
+`somefile.nix`
+
+~~~nix
+{
+  /**
+  Documentation for mapAttrs
+  
+  # Example
+  
+  ```
+  mapAttrs {attr = "foo"; } ...
+  ```
+  
+  # Type
+  
+  ```
+  mapAttrs :: a -> b -> c
+  ```
+  
+  */
+  mapAttrs = f: s: <...>;
+}
+~~~
+
+## Examples
+
+This section contains examples for the different use cases we would end up with; Visualize them and emphasize the previously discussed characteristics.
+
+### Attribute bindings 
+
+`somefile.nix`
+
+~~~nix
```

Comment:
```suggestion
```nix
```

Reply:
problem is that I used ``` for the examples within the nix code block. Markdown has the ~~~ variant which allows usage of ``` within. Doesn't this render correctly on your editor?

Reply:
Note that if you need to nest code blocks, you can use more than 3 backticks for the outer block, like:
`````
````foo
And an inner block:
```bar
Inner
```
Back to outer block
````
`````

Reply:
Thanks. I didn't know that. Changed it then. 

Noob question: What is the reason for not using ~~~?

Reply:
I guess it's another way yes, I honestly never used it, and rarely see it used either

Reply:
I wasn't aware ~~~ existed at all. The "more than 3 backticks" solution sounds nicer IMHO